### ISSUE TITLE
Memory leak when using gui.Application with scene widget to reproduce animation of mesh

### ISSUE THREAD

**Issue by xabirizar9** [isl-org/Open3D#4341]
Similar issues: #1239

**Describe the bug**
When visualizing big triangle meshes (as an example, see bunny STL model) using gui.Application together with gui.SceneWidget,  applying transformations to the meshes and accorindgly removing and adding the geometry leads to the program completely crashing the computer. The RAM memory fills up in seconds. I tried some solutions proposed in other posts but they are not working.

**To Reproduce**

1. Save "064_Bunny01.stl" model to the same directory where the script is running
2. Run script (in terminal): python visualize_environment.py 
3. In a second terminal window, run "htop" to check RAM usage

**Expected behavior**
In each iteration, the scene should be reloaded normally without making copies of itself and running out of memory

**Environment (please complete the following information):**

- Operating system: Ubuntu 20.04.3 LTS
- Python version: Python 3.8.10
- Open3D version: 0.13
- Is this remote workstation?: NO
- How did you install Open3D?: pip

**Additional context**

Usage of remove_geometry() or clear_geometry() seems to make no difference.
On each iteration, I remove the geometry, apply a rotation, and add the geometry again.

```python
#!/usr/bin/env python
import numpy as np
import open3d as o3d
from open3d.visualization import gui


class WindowScene:
    def __init__(self):
        self.window = gui.Application.instance.create_window("Scene visualization", 1920, 1080)

        self.object = o3d.io.read_triangle_mesh("064_Bunny01.stl", enable_post_processing=True)
        self.default_material = o3d.visualization.rendering.Material()
        self.default_material.shader = "defaultLit"
        self.int_value = 0

        self.scene_widget = gui.SceneWidget()
        self.scene_widget.scene = o3d.visualization.rendering.Open3DScene(self.window.renderer)

        self.scene_widget.scene.scene.set_sun_light([-1, -1, -1], [1, 1, 1], 20000)  # Direction, Color, Intensity
        self.scene_widget.scene.scene.enable_sun_light(True)

        self.bbox = o3d.geometry.AxisAlignedBoundingBox([-10, -10, -10], [10, 10, 10])
        self.scene_widget.setup_camera(60, self.bbox, [0, 0, 0])

        self.window.add_child(self.scene_widget)
        self.window.set_on_tick_event(self.on_tick_event)

    def on_tick_event(self):
        self.int_value += 1
        self._on_slider(self.int_value)

    def _on_slider(self, scene_index):
        self.scene_widget.scene.remove_geometry(str(id(self.object)))
        x = scene_index/10000
        rotation = np.array([[1,         0,          0, 0],
                             [0, np.cos(x), -np.sin(x), 0],
                             [0, np.sin(x),  np.cos(x), 0],
                             [0,         0,          0, 1]], dtype=np.float64)
        self.object.transform(rotation)
        self.scene_widget.scene.add_geometry(str(id(self.object)), self.object, self.default_material)


def main():
    gui.Application.instance.initialize()

    w = WindowScene()

    gui.Application.instance.run()


if __name__ == "__main__":
    main()
```
[064_Bunny01.zip](https://github.com/isl-org/Open3D/files/7596673/064_Bunny01.zip)

**Comment by mag-sruehl** [comment]
Dear open3d community,
I am working on the same project as @xabirizar9 and have looked more into the issue. While I could not figure out why add_geometry/remove_geometry has growing memory consumption as show in the example, I came to the conclusion that this is not our primary issue. Instead, for what we want to do, there is no need to call add_geometry more than once. 

Instead, we must only move the orgin of the mesh. The FilamentScene offers the method SetGeometryTransform to do that, but it is not implemented in Open3DScene and thus not exposed to the python api. I have created a pull request https://github.com/isl-org/Open3D/pull/4366 which fixes this issue

**Comment by errissa** [comment]
Thank you @xabirizar9  and @mag-sruehl for reporting this issue. You are correct that you only need to update the transform of the object rather than adding/removing it with each change, but as you noted, that capability was not exposed in Python. Thank you for your contribution. I have reviewed and commented on PR #4366.

A couple of further comments:

1. There is a memory leak issue with `add_geometry` which is resolved by PR #4421.
2. The `on_tick_event` function should end with `self.window.post_redraw()`. Without that, the screen will not redraw unless something forces it to (e.g., an exposure event, mouse movement). So, with each tick the work of adding/removing geometry was being queued up to Filament but not being processed which made the memory leak look worse; memory was allocated but not deallocated until Filament got a chance to process the queued work.

**Comment by errissa** [comment]
PR #4366 and #4421 are both now merged.
